### ISSUE TITLE
Ability to Restrict Types of Blocks Shown when Child Blocks are Put into a Tab

### ISSUE THREAD

**Issue by brianrivet-tilt** [spicywebau/craft-neo#513]
This is a feature request. I was wondering if it might be possible to set up the feature where you can put the child blocks for a block into a tab so that you can limit the types of child blocks that can go into the tab. Let me give you my use case as that might explain it better:

Let's say I want to set up a Neo field as a page builder, but rather than have a bunch of fields on my content blocks that control things like styling, I would like to have a block type that can be a child on different blocks that is a style customization block, and it can have different children for different types of style customizations with all the fields needed to do the customization. Then I'd like to have a different set of child blocks for actual content children (so for example if I have a block that is a column, be able to have child blocks that are for content like text, images, etc. but then also have a child block available that would let me add style customizations). Then I'd like to be able to put in two tabs on parent block, one for content children, and one for style customizations. Both tabs would be listing child blocks, but the tab with style customizations would only list child blocks that I had marked as being allowed to show in that list, and the content tab would only show blocks I allowed in that list. All of the blocks are children of the current block, but they get shown in a different list depending on how I set the child lists as far as what they can show.

The thinking is that it would allow the user to add style customizations only when needed without having the overhead of all the styling fields on all the different blocks.

**Comment by turnstylerj** [comment]
Another handy use case for this would be a block with repeatable "row" child blocks where the author could add as many as they'd like/reorder them, but also child block(s) for a "Call to Action" type button that maybe has a couple of block options but is always intended to be used below/separate from the repeatable row blocks.

**Comment by ttempleton** [comment]
There are no current plans to implement this, but just putting this here so I don't forget, in case this is worked on in the future.

Per the discussion in #753, if this were to be implemented, the idea could be expanded beyond restricting different child block types to different tabs, to allow different child block types to be restricted to different places in the same tab.

**Comment by sander-lameco** [comment]
Yes a solution would be to allow multiple Child Block Fields`UI element`, and add Child Block limitations/conditions there?
